### DISCUSSION TITLE
Migrating from old PDEBuild to Tycho headless build

### DISCUSSION THREAD

**Discussion by Dani-Hub** [eclipse-tycho/tycho#4554] [Q&A]
Hi, we need to migrate our existing headless PDEBuild RCP applications by a the tycho build. In our situation we have *all* or plugins already build and they are stored in our maven repo. We don't want to rebuild them. Similarly, all our features and the product file are accessible as maven artifacts. The old PDEBuild was able to grab all these things and to create the executable including the products plugins and feature and p2 folder.

How can I use a headless tycho build to do just this bare minimum of work? Ideally we would just like to reuse our existing features and thirdparty bundles (that is the Eclipse RCP plugins for a dedicated Eclipse version that we also have stored in our artifactory). Is this possible and if so how?

I read the part about the [recommended layout](https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html), but my impression is that the bundles subfolder exists for projects to be build. But I don't want to build them, we have already build them as OSGi compatible jars. We only want that the PDEBuild replacement is able to honor the possible feature.xml directives that say which bundles need to to be unpacked and which not. In PDEBuild it was possible to specify a folder for everything pre-build (The thirdparty folder). How can I specify such a "thirdparty" folder for things that have been prebuild? (As I said: In our case we have *all* bundles rebuild, we just want to invoke a process that creates the final executable including the folder structure for plugins, features, and p2 directory.

**Comment by laeubi** [comment]
Tycho supports using a folder of jars with target location type "Directory", beside that you can even use "Maven" location type both are also supported in PDE inside the IDE so you should get a smooth integration experience.

So you just can use the [recommended layout](https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html), leave out whatever you don't need e.g. just using a pde target file and the features folder. The you just need the configuration pom and the rest is just plain PDE.

Beside that, if it is crucial to your business you can find dedicated support options for Tycho here:
https://github.com/eclipse-tycho/tycho#getting-support

**Comment by Dani-Hub** [comment]
Hi @laeubi : Thanks for your response. This gives me an initial hint in regard to the target location directory. Does this target location directory also allow to specify a `features` folder which contains the collections of all features? We have all of these provided as maven artifacts as well, so it would be very inconvenient for us to restructure them into individual <modules> entries. Thanks very much!
